internal/configuration/command.go

```diff
@@ -60,39 +62,46 @@ func getConfigWhiteList(cfg *config.Config) map[string]*configInfo {
 		field := t.Field(i)
 		// currently only boolean fields are part of this whitelist, but this may change in the future
 		if kind := field.Type.Kind(); kind == reflect.Bool {
-			if jsonFieldName := getJsonFieldName(field, cfg.IsTest); jsonFieldName != "" {
+			if jsonFieldName := getJsonFieldName(field); jsonFieldName != "" {
 				whitelist[jsonFieldName] = &configInfo{
-					kind: kind,
-					name: field.Name,
+					kind:     kind,
+					name:     field.Name,
+					readOnly: isReadOnly(jsonFieldName, cfg.IsTest),
 				}
 			}
 		}
 	}
 	return whitelist
 }
 
-func getJsonFieldName(field reflect.StructField, isTest bool) string {
+func getJsonFieldName(field reflect.StructField) string {
 	jsonTag := field.Tag.Get("json")
 	if jsonTag == "-" {
 		return ""
 	}
 	if strings.Contains(jsonTag, ",") {
 		jsonTag, _, _ = strings.Cut(jsonTag, ",")
 	}
+	// Want to hide this from linux and mac until the name specifies this field only affects Windows to avoid confusion
 	if jsonTag == "disable_plugins_once" && runtime.GOOS != "windows" {
 		return ""
 	}
-	if jsonTag == "disable_updates" && !isTest && update.IsHomebrew() {
-		return ""
-	}
 	return jsonTag
 }
 
+func isReadOnly(jsonField string, isTest bool) bool {
+	if jsonField == "disable_updates" && !isTest && update.IsHomebrew() {
+		return true
+	}
+	return false
```

Comment:
Why do we include `!isTest` in here?
```suggestion
	return jsonField == "disable_updates" && !isTest && update.IsHomebrew()
```

Reply:
The `IsHomebrew()` function is quite slow, and causes the integration tests to time out because the whitelist is generated as part of the `command` struct, which is called in every test. However, I shouldn't have `isTest` here anymore.

The whitelist should instead be generated lazily (only when using the `config` subcommands), rather than whenever `New()` is called to create these commands (which is always).